Code/GraphMol/Depictor/Templates.h

```diff
@@ -46,23 +46,99 @@ class CoordinateTemplates {
 
- private:
-  CoordinateTemplates() {
-    // load templates into m_templates map by atom count
+  bool setRingSystemTemplates(const std::string& template_dir) {
+    std::ifstream cxsmiles(template_dir);
+    if (!cxsmiles) {
+      BOOST_LOG(rdWarningLog)
+          << "WARNING: Could not open file " << template_dir << std::endl;
+      return false;
+    }
+
+    // Try loading templates in from this directory, if unsuccessful, keep current templates
+    std::unordered_map<unsigned int, std::vector<std::shared_ptr<RDKit::ROMol>>> templates;
+    std::string line;
+    bool success = true;
+    while (std::getline(cxsmiles, line)) {
+      RDKit::ROMol* mol_ptr = RDKit::SmilesToMol(line);
+      if (!mol_ptr) {
+        BOOST_LOG(rdWarningLog)
+            << "WARNING: Could not load templates from " << template_dir
+            << ": Invalid smiles" << std::endl;
+        success = false;
+        break;
+      }
+      std::shared_ptr<RDKit::ROMol> mol(mol_ptr);
+      // check if mol has coordinates -- 3D coordinates are OK
+      if (mol->getNumConformers() == 0) {
+        BOOST_LOG(rdWarningLog)
+            << "WARNING: Could not load templates from " << template_dir
+            << ": One or more templates are missing coordinates\n";
+        success = false;
+        break;
+      }
+      // template must be a ring system
+      if (!isRingSystem(*mol)) {
+        BOOST_LOG(rdWarningLog)
+            << "WARNING: Could not load templates from " << template_dir
+            << ": One or more templates are not a single ring system\n";
+        success = false;
+        break;
+      }
+      templates[mol->getNumAtoms()].push_back(mol);
+    }
+    cxsmiles.close();
+    if (success) {
+      clearTemplates();
+      m_templates = std::move(templates);
+    }
+    return success;
+  }
+
+  void loadDefaultTemplates() {
+    clearTemplates();
+    // load default templates into m_templates map by atom count
     for (const auto& smiles : TEMPLATE_SMILES) {
         std::shared_ptr<RDKit::ROMol> mol(RDKit::SmilesToMol(smiles));
         m_templates[mol->getNumAtoms()].push_back(mol);
     }
   }
 
+ private:
+  CoordinateTemplates() {
+    loadDefaultTemplates();
+  }
   CoordinateTemplates(const CoordinateTemplates&) = delete;
   CoordinateTemplates& operator= (const CoordinateTemplates&) = delete;
-  ~CoordinateTemplates() {
+
+  void clearTemplates() {
     for (auto& [atom_cout, romols] : m_templates) {
       romols.clear();
     }
     m_templates.clear();
   }
 
+  bool isRingSystem(RDKit::ROMol& mol) {
+    // Make sure this template is a ring system (spiro'd ring systems are OK).
+    // We can check this by ensuring that every atom is in a ring and that there
+    // is only one connected component in the molecular graph.
+    if (RDKit::MolOps::getMolFrags(mol).size() != 1) {
+      return false;
+    }
+
+    // Use symmetrizeSSSR since that is what is used in coordinate generation
+    RDKit::MolOps::symmetrizeSSSR(mol);
```

Comment:
I'd do:

```
bool includeDativeBonds = true;
RDKit::MolOps::symmetrizeSSSR(mol, includeDativeBonds);
```

So that we don't get trouble for organometallics (Greg recently added this).

Also, we don't run this check on the default templates, do we?

Reply:
Well, maybe we could add this check to the default templates then in a C++ test just verify the templates are loaded then also check bond lengths

Reply:
Ok, that is what I ended up doing :)